packages/drivers/odsp-driver/src/graph.ts

```diff
@@ -49,7 +83,41 @@ export async function getShareLink(
     driveId: string,
     itemId: string,
     identityType: IdentityType,
-    logger?: ITelemetryLogger,
+    logger: ITelemetryLogger,
+    scope: "anonymous" | "organization" | "default" | "existingAccess" = "existingAccess",
+    type: "view" | "edit" = "edit",
+    msGraphOrigin?: string,
+): Promise<string | undefined> {
+    let result: string | undefined;
+    let success = false;
+    let retryAfter = 0;
+    do {
+        try {
+            result = await getShareLinkCore(getShareLinkToken, siteUrl, driveId, itemId, identityType,
+                logger, scope, type, msGraphOrigin);
+            success = true;
+        } catch (err) {
+            // If it is not retriable, then just throw the error.
+            if (!canRetryOnError(err)) {
+                throw err;
+            }
+            // If the error is throttling error, then wait for the specified time before retrying.
+            // If the waitTime is not specified, then we start with retrying immediately to max of 8s.
+            retryAfter = getRetryDelayFromError(err) ?? Math.min(retryAfter * 2, 8000);
```

Comment:
This can take forever. Are users of this API aware of it? I'd put a big note in the header for this function.
Ideally we should start designing these APIs properly, i.e. by passing cancelation token into such functions, and checking if caller wants to cancel it. This is probably for next change, so mostly FYI

Reply:
Added comment for now. Makes sense.